lib/time.js

```diff
@@ -349,6 +415,219 @@ function CronTime(luxon) {
 			return date;
 		},
 
+		/**
+		 * Search backwards in time 1 minute at a time, to detect a DST forward jump.
+		 * When the jump is found, the range of the jump is investigated to check for acceptable cron times.
+		 * A pair is returned, whose first is a boolean representing if an acceptable time was found inside the jump,
+		 * and whose second is a DateTime representing the first millisecond after the jump.
+		 *
+		 * The input date is expected to be decently close to a DST jump.
+		 * Up to a day in the past is checked before an error is thrown.
+		 * @param date
+		 */
+		_findPreviousDSTJump: function (date) {
+			/** @type number */
+			let expectedMinute, expectedHour, actualMinute, actualHour;
+			/** @type DateTime */
+			let maybeJumpingPoint;
+
+			// representing one day of backwards checking. If this is hit, the input must be wrong.
+			const iterationLimit = 60 * 24;
+			let iteration = 0;
+			do {
+				if (iteration++ > iterationLimit) {
+					throw new Error(
+						`ERROR: This DST checking related function assumes the input DateTime (${date.toISO()}) is within 24 hours of a DST jump.`
+					);
+				}
+
+				expectedMinute = date.minute - 1;
+				expectedHour = date.hour - 1;
```

Comment:
it's a little confusing why the minute and hour are both being decremented here, especially if we're checking every minute. could you add some comments explaining why?

Reply:
It's confusing because it makes no sense! I have no idea what I was thinking there. Not to mention the `if hour < 0, hour += 60` after. What??

The only reason it didn't bug out in unit tests, is because it happens to still work for 'full' hour jumps. It would completely break for the +30 minute jump though.

Also, there is no loop variable? After fixing this first issue, I find out this is just an infinite loop if the first iteration doesn't have the DST jump. Which it previously did not have due to going backwards by an hour incorrectly. 

In short, this code was very broken, and I am impressed this somehow passed all tests, especially the iteration limit test.

Reply:
glad you got it figured out!